spec/modules/helpers.rb

```diff
@@ -0,0 +1,88 @@
+module SessionHelpers
+  def create_user(role)
+    User.create(
+      login: OmniAuth.config.mock_auth[:google_oauth2][:info][:name],
+      provider: OmniAuth.config.mock_auth[:google_oauth2][:provider],
+      uid: OmniAuth.config.mock_auth[:google_oauth2][:uid],
+      auth_role: role,
+    )
+  end
+
+  def login_as_user
+    create_user("user")
+    visit root_path
+    click_on "Login"
+    expect(page).to have_text("Logged in!")
+  end
+  
+  def login_as_moderator
+    create_user("moderator")
+    visit root_path
+    find_button "Login"
+    click_on "Login"
+    expect(page).to have_text("Logged in!")
+  end
+
+  def login_as_admin
+    create_user("admin")
+    visit root_path
+    click_on "Login"
+    expect(page).to have_text("Logged in!")
+  end
+
+  def logout
+    visit root_path
+    click_on "Logout"
+    expect(page).to have_text("Logged out!")
+  end
+end
+
+module CategoryHelpers
+  def create_category(category_name, category_description)
+    login_as_admin
+    find_button "New Category"
+    click_on "New Category"
+    find_button "Submit"
+    fill_in("Name", with: category_name)
+    fill_in("Description", with: category_description)
+    click_on "Submit"
+    should_be_on_categories_page(category_name, category_description)
+  end
+
+  def should_be_on_categories_page(category_name, category_description, action = "created")
+    expect(page).to have_text("Category was successfully #{action}.")
+    expect(page).to have_text("Categories > #{category_name}")
+    expect(page).to have_text(category_name)
+    expect(page).to have_text(category_description)
+    find_button "Edit"
+    find_button "Create Discussion"
+    expect(page).to have_text("DISCUSSION REPLIES LAST POST")
+  end
```

Comment:
This looks like you're building an RSpec matcher, since you're not mutating the page, only asserting based upon what's present.  What do you think about defining it as an [actual RSpec matcher?](https://relishapp.com/rspec/rspec-expectations/v/2-3/docs/custom-matchers/define-matcher)

Reply:
Done => [0510d60](https://github.com/Brian-Demon/conclave/pull/30/commits/0510d60206730ed8d82ede54caab9c00daa2849f)